witchcraft-log/src/logger.rs

```diff
@@ -47,20 +47,32 @@ impl Log for NopLogger {
     fn flush(&self) {}
 }
 
-static LOGGER: AtomicLazyCell<&'static dyn Log> = AtomicLazyCell::NONE;
+static LOGGER: OnceLock<&'static dyn Log> = OnceLock::new();
 
-/// Sets the global logger.
+/// Sets the global logger to a `&'static dyn Log`.
 ///
 /// The global logger can only be set once. Further calls will return an error.
 pub fn set_logger(logger: &'static dyn Log) -> Result<(), SetLoggerError> {
-    LOGGER.fill(logger).map_err(|_| SetLoggerError(()))
+    LOGGER.set(logger).map_err(|_| SetLoggerError(()))
+}
+
+/// Sets the global logger to a `Box<dyn Log>`.
+///
+/// The global logger can only be set once. Further calls will return an error.
+pub fn set_boxed_logger(logger: Box<dyn Log>) -> Result<(), SetLoggerError> {
+    let mut logger = Some(logger);
+    LOGGER.get_or_init(|| Box::leak(logger.take().unwrap()));
+    match logger {
+        Some(_) => Err(SetLoggerError(())),
+        None => Ok(()),
+    }
 }
```

Comment:
Adding this because it's annoying to avoid leaking the logger if one's already been installed otherwise.